src/tests/ftest/util/macsio_util.py

```diff
@@ -0,0 +1,486 @@
+        #       reported in powers of 1000 and unit symbols are Kb, Mb, Gb, Tb,
+        #       Pb.  See http://en.wikipedia.org/wiki/Binary_prefix. for more
+        #       information
+        self.units_prefix_system = FormattedParameter(
+            "--units_prefix_system {}")
+
+        #   --interface %s [miftmpl]
+        #       Specify the name of the interface to be tested. Use keyword
+        #       'list' to print a list of all known interface names and then
+        #       exit.
+        self.interface = FormattedParameter("--interface {}", "hdf5")
+
+        #   --parallel_file_mode %s %d [MIF 4]
+        #       Specify the parallel file mode. There are several choices.  Use
+        #       'MIF' for Multiple Independent File (Poor Man's) mode and then
+        #       also specify the number of files. Or, use 'MIFFPP' for MIF mode
+        #       and one file per processor or 'MIFOPT' for MIF mode and let the
+        #       test determine the optimum file count. Use 'SIF' for SIngle
+        #       shared File (Rich Man's) mode. If you also give a file count for
+        #       SIF mode, then MACSio will perform a sort of hybrid combination
+        #       of MIF and SIF modes.  It will produce the specified number of
+        #       files by grouping ranks in the the same way MIF does, but I/O
+        #       within each group will be to a single, shared file using SIF
+        #       mode.
+        #
+        #       Run macsio with SIF mode. MIF mode uses the HDF5 posix driver,
+        #       so it won't go through MPI-IO and hence not through the MPI-IO
+        #       DAOS driver.
+        #
+        #       Note: Value should be specified as a string of a space-
+        #           separated string and integer value, e.g. 'SIF 1'.
+        self.parallel_file_mode = FormattedParameter(
+            "--parallel_file_mode {}", "SIF 1")
+
+        #   --avg_num_parts %f [1]
+        #       The average number of mesh parts per MPI rank. Non-integral
+        #       values are acceptable. For example, a value that is half-way
+        #       between two integers, K and K+1, means that half the ranks have
+        #       K mesh parts and half have K+1 mesh parts. As another example,
+        #       a value of 2.75 here would mean that 75% of the ranks get 3
+        #       parts and 25% of the ranks get 2 parts. Note that the total
+        #       number of parts is this number multiplied by the MPI
+        #       communicator size. If the result of that product is
+        #       non-integral, it will be rounded and a warning message will be
+        #       generated.
+        self.avg_num_parts = FormattedParameter("--avg_num_parts {}")
+
+        #   --mesh_decomp %d %d %d []
+        #       The layout of parts in the mesh overriding the simple
+        #       decomposition e.g. 4 8 1 will decompose into 32 parts in the
+        #       stucture (x y z).
```

Comment:
(style)  'stucture' may be misspelled - perhaps 'structure'?

Reply:
Fixed